np.asarray(PandasSeries), with default `copy=None` returns read-only array (pandas 3.0.0)

As reported in [this comment](https://github.com/stumpy-dev/stumpy/pull/1117#issuecomment-3782269359), [some tests are failing](https://github.com/stumpy-dev/stumpy/actions/runs/21233930418). 

The error shows:

```
# in core.preprocess_non_normalized

T[~np.isfinite(T)] = np.nan
        ^^^^^^^^^^^^^^^^^^
E       ValueError: assignment destination is read-only
```

And this happens when the input that is passed to `core.preprocess_non_normalized` is pandas.Series. This behaviour can be reproduced by using the following MWE:

```
import pandas as pd 
import numpy as np

print(pd.__version__)  # 3.0.0
print(np.__version__)  # 2.3.5


T_B = np.array([ 584.,  -11.,   23.,   79., 1001.,    0.,  -19.])
ps = pd.Series(T_B)
T = np.asarray(ps)
T[0] = 100
```

Further investigation shows that `np.shares_memory(T, ps.array)` returns True, and and according to [read-only array section in pandas Copy-on-Write (CoW)](https://pandas.pydata.org/docs/dev/user_guide/copy_on_write.html#read-only-numpy-arrays):

> Accessing the underlying NumPy array of a DataFrame will return a read-only array if the array shares data with the initial DataFrame:

--- a/stumpy/core.py
+++ b/stumpy/core.py
@@ -2117,8 +2117,9 @@ def _preprocess(T, copy=True):
         Time series or sequence
 
     copy : bool, default True
-        A boolean value that indicates whether the process should be done on
-        input `T` (False) or its copy (True).
+        A boolean value that indicates whether the process should be performed on
+        input array `T` (False) or its copy (True). If `T` is a dataframe, then the
+        data is always copied into a brand new array.
 
     Returns
     -------
@@ -2173,8 +2174,9 @@ def preprocess(
         Window size
 
     copy : bool, default True
-        A boolean value that indicates whether the process should be done on
-        input `T` (False) or its copy (True).
+        A boolean value that indicates whether the process should be performed on
+        input array `T` (False) or its copy (True). If `T` is a dataframe, then the
+        data is always copied into a brand new array.
 
     M_T : numpy.ndarray, default None
         Rolling mean
@@ -2236,8 +2238,9 @@ def preprocess_non_normalized(T, m, copy=True):
         Window size
 
     copy : bool, default True
-        A boolean value that indicates whether the process should be done on
-        input `T` (False) or its copy (True).
+        A boolean value that indicates whether the process should be performed on
+        input array `T` (False) or its copy (True). If `T` is a dataframe, then the
+        data is always copied into a brand new array.
 
     Returns
     -------
@@ -2296,8 +2299,9 @@ def preprocess_diagonal(
         corresponding value set to False in this boolean array.
 
     copy : bool, default True
-        A boolean value that indicates whether the process should be done on
-        input `T` (False) or its copy (True).
+        A boolean value that indicates whether the process should be performed on
+        input array `T` (False) or its copy (True). If `T` is a dataframe, then the
+        data is always copied into a brand new array.
 
     Returns
     -------

